docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,288 @@
+
+```bash
+$ oras repo tags layout-dir --oci-layout
+
+v1-linux-amd64 
+v1-linux-arm64
+v1-linux-armv7
+```
+
+2. Create a multi-arch image by combining two image manifests into an image index, tag it with `v1` and push the tagged image index to an OCI image layout `layout-dir` automatically. Add annotations to this image index at the same time:  
+
+```console
+$ oras manifest index create layout-dir:v1 v1-linux-amd64 v1-linux-arm64 --oci-layout --annotation "com.example.key=value" 
+
+Fetching  v1-linux-amd64 
+Fetched   sha256:42c524c48e0672568dbd2842d3a0cb34a415347145ee9fe1c8abaf65e7455b46 v1-linux-amd64 
+Fetching  v1-linux-arm64
+Fetched   sha256:965945e1a08031a63d5970c1da7c39af231c36e4c0a5a3cc276d02a3e06513ee v1-linux-arm64 
+Packed    edb5bc1f0b5c application/vnd.oci.image.index.v1+json
+Pushed    [oci-layout] layout-dir:v1
+Digest: sha256:edb5bc1f0b5c21e9321b34e50c92beae739250fb88409056e8719d9759f6b5b4
+
+Status: An image index has been created and pushed to layout-dir:v1
+```
+
+3. View the image index in the OCI image layout in a pretty JSON output: 
+
+```bash
+$ oras manifest fetch layout-dir:v1 --oci-layout --pretty  
+```
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:42c524c48e0672568dbd2842d3a0cb34a415347145ee9fe1c8abaf65e7455b46",
+      "size": 1239,
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      }
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:965945e1a08031a63d5970c1da7c39af231c36e4c0a5a3cc276d02a3e06513ee",
+      "size": 1239,
+      "platform": {
+        "architecture": "arm64",
+        "os": "linux"
+      }
+    }
+  ],
+  "annotations": {
+    "com.example.key": "value"
+  }
+}
+```
+
+#### Create a multi-arch image from a registry only
+
+Create a multi-arch image using two arch-specific images stored in a registry only, tag it with `v1` and push the tagged image index to a registry automatically.
+
+```console
+$ oras manifest index create localhost:5000/example:v1 v1-linux-arm64 v1-linux-arm64
+
+Fetching  v1-linux-amd64 
+Fetched   sha256:42c524c48e0672568dbd2842d3a0cb34a415347145ee9fe1c8abaf65e7455b46 v1-linux-amd64 
+Fetching  v1-linux-arm64
+Fetched   sha256:965945e1a08031a63d5970c1da7c39af231c36e4c0a5a3cc276d02a3e06513ee v1-linux-arm64 
+Packed    edb5bc1f0b5c application/vnd.oci.image.index.v1+json
+Pushed    localhost:5000/example:v1
+Digest: sha256:edb5bc1f0b5c21e9321b34e50c92beae739250fb88409056e8719d9759f6b5b4
+```
+
+#### Create a multi-arch image from local and registry
+
+Users need to pull the target arch-specific image to the local repository or OCI image layout, then use `oras manifest index create` to create a multi-arch image.  
```

Comment:
Do we want to demonstrate the example commands for this scenario?

Reply:
Added examples